graylog2-server/src/main/java/org/graylog2/indexer/cluster/jest/JestUtils.java

```diff
@@ -74,8 +75,15 @@ private static ElasticsearchException specificException(Supplier<String> errorMe
 
         for (JsonObject rootCause : rootCauses) {
             final String type = asString(rootCause.get("type"));
-            if ("query_parsing_exception".equals(type)) {
-                return buildQueryParsingException(errorMessage, rootCause, reasons);
+            if (type == null) {
+                continue;
+            }
+            switch(type) {
+                case "query_parsing_exception":
+                    return buildQueryParsingException(errorMessage, rootCause, reasons);
+                case "index_not_found_exception":
+                    final String indexName = asString(rootCause.get("resource.id"));
+                    return new IndexNotFoundException("Index not found for query: " + indexName + ". Try recalculating your index ranges.");
```

Comment:
What do you think about using `errorMessage` here, too? This would give developers the possibility to customize the error message (also see [`Counts#totalCount()`](https://github.com/Graylog2/graylog2-server/blob/a22a712411262a676ce1d0a9cfdfc969f5a48a88/graylog2-server/src/main/java/org/graylog2/indexer/counts/Counts.java#L67)